### PR TITLE
Adding update functionality to the operator

### PR DESCRIPTION
**Pull request by RameshaKaru**
User can update the api using the following command and a kubectl extension and operator will handle the request to update the deployment.

`kubectl update api api-name --from-file=<filename> --replicas=<replicas> -n=<preferred namespace>`

Also, the api kind structure was modified to a user-friendly way
Further, this PR allows the user to give his preferred namespace using the kubectl add api and kubectl update api commands, instead of giving it from the controller config file. The api cr object, microgateway deployment and service will be created in the user given namespace.

### CODE REVIEW

File: apim-operator/deploy/kubectl-extension/kubectl-add.sh [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#!/bin/bash
```
Comment (dinusha92) [inline apim-operator/deploy/kubectl-extension/kubectl-add.sh]: Shall we add the license header here

File: apim-operator/pkg/controller/api/api_controller.go [hunk 2]
```diff
@@ -416,16 +414,63 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
...
 
 	svc := createMgwLBService(instance, userNameSpace)
 	svcFound := &corev1.Service{}
 	svcErr := r.client.Get(context.TODO(), types.NamespacedName{Name: svc.Name, Namespace: svc.Namespace}, svcFound)
 
-	//Schedule Kaniko pod
-	if imageExist {
+	if instance.Spec.UpdateTimeStamp != "" {
+		//Schedule Kaniko pod
+		log.Info("Update API")
```
Comment (dinusha92) [inline apim-operator/pkg/controller/api/api_controller.go]: Can we log which API we are updating here. It's better if the log is more descriptive

File: apim-operator/deploy/kubectl-extension/kubectl-add.sh (hunk 1 above)
Comment (dinusha92) [reply apim-operator/deploy/kubectl-extension/kubectl-add.sh]: Sorry I've missed. The license is already here :)

File: apim-operator/pkg/controller/api/api_controller.go (hunk 2 above)
Comment (RameshaKaru) [reply apim-operator/pkg/controller/api/api_controller.go]: Added the logs

**Pull request merged**